src/index.js

```diff
@@ -188,16 +189,17 @@ async function replicate({ seq }) {
             seq: docs[docs.length - 1].seq,
           })
         )
-        .then(({ seq: lastDocSeq }) => {
-          if (lastDocSeq >= npmSeqToReach) {
-            log.info('🐌 We reached the npm current sequence');
-            changes.cancel();
-          }
-        })
-        .then(done)
+        .then(() => done())
```

Comment:
this could stay `.then(done)`

Reply:
No, because then err would be triggered, look at the call back supplied to the queue

Reply:
Not sure if I get the difference, this both is the same:

```js
Promise.resolve('hi').then(console.log);
// identical
Promise.resolve('hi').then(msg => console.log(msg));
```

Reply:
The callback you supply to the queue is (`done`):
```
if (err) {
  reject(err)
}
```
So if there is any argument given to the function `done` it is going to throw an error.

```
.then(() =>
          stateManager.save({
            seq: docs[docs.length - 1].seq,
          })
        )
        .then(done)
        .catch(done);
```

If you did it this way, `stateManager.save` if going to resolve with `state`, and that `state` is going to get passed to the `done` function as an argument and `done`, because it just does `if (err)`, is going to throw an error.

Reply:
ah okay, good call, thanks